src/VectorStore.ts

```diff
@@ -320,14 +303,25 @@ export class VectorStore {
 		return new Map(dbFile.embeddings)
 	}
 
-	getNearestVectors(searchVector: Vector, resultNumber: number, relevanceThreshold: number): NearestVectorResult[] {
+	private similarityBatch(searchVectors: Vector[], possibleMatch: Vector): number {
+		let highestSimilarity = similarity(searchVectors.first(), possibleMatch)
+		for (const searchVector of searchVectors) {
+			const currentSimilarity = similarity(searchVector, possibleMatch)
+			if (currentSimilarity > highestSimilarity) {
+				highestSimilarity = currentSimilarity
+			}
+		}
+		return highestSimilarity
+	}
+
+	getNearestVectorsBatch(searchVectors: Vector[], resultNumber: number, relevanceThreshold: number): NearestVectorResult[] {
```

Comment:
nit: IMO we're using the word batch in a lot of different contexts, I think it's ok to just name this `getNearestVectors`